pallets/delegation/src/lib.rs

```diff
@@ -387,6 +439,10 @@ pub mod pallet {
 				Error::<T>::UnauthorizedDelegation
 			);
 
+			CurrencyOf::<T>::reserve(&payer, deposit_amount)?;
```

Comment:
M(reserve-deposit part 2): same here, i would reserve inside the `store_delegation_under_parent`